jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java

```diff
@@ -62,6 +63,7 @@ public AbstractReportGenerator createGenerator() {
         return generator;
     }
 
+    @Internal
```

Comment:
I think you are right, but I am not 100% sure.

Reply:
After A LOT of digging, I arrived at the following conclusion from the list of annotations [Task property annoations](https://docs.gradle.org/current/userguide/more_about_tasks.html) only `@Input` ,`@Console`,  `@Internal`,  `@Nested` and `@ReplacedBy` are even allowed by the type system. (all other annotations refer to files)
* `@ReplacedBy` makes no sense, there is no property that could replace it
* `@Nested` could be used, but that would defer the issue to the `private final String text` member of the `enum` that is returned.  That however is only used inside the enum.
* `@Input` the value would have to be writeable for that to make sense, but there is no setter.  I actually tried to set the value in a test. Fails miserably.
* `@Console` a value that affects the console output. Like `@Input` I don't see this as a viable opion, because the property cannot be written.

Consequently the only annotion that is appropriate is `@Internal`

Reply:
I decided against persisting the test, because I see little point in having a test try to set a property that you're not supposed to set.